README.md

```diff
@@ -5,757 +5,886 @@
+  * [chaining transitions](#chaining-transitions)
+  * [initialize transition](#initialize-transition)
+  * [set transition](#set-transition)
+  * [Scope rules](#scope-rules)
+* [microstates npm package](#microstates-npm-package)
+  * [create(Type, value): Microstate](#createtype-value-microstate)
+  * [from(any): Microstate](#fromany-microstate)
+  * [map(fn: tree =&gt; tree, microstate: Microstate): Microstate](#mapfn-tree--tree-microstate-microstate-microstate)
+* [Middleware](#middleware)
+* [Observable Microstates](#observable-microstates)
+* [The Vision of Microstates](#the-vision-of-microstates)
+  * [Shared Solutions](#shared-solutions)
+  * [Added Flexibility](#added-flexibility)
+  * [Framework Agnostic Solutions](#framework-agnostic-solutions)
 * [FAQ](#faq)
   * [What if I can't use class syntax?](#what-if-i-cant-use-class-syntax)
   * [What if I can't use Class Properties?](#what-if-i-cant-use-class-properties)
 * [Run Tests](#run-tests)
 
-## Intro
+<!-- tocstop -->
+</details>
 
-Microstates are a _typed_, _composable_ and _immutable_ state container.
+# Features
 
-Let's get some terminology out of the way so we're on the same page for the rest of the README.
+With microstates added to your project, you get:
 
-* _Typed_ means that microstates know about the kind of data it contains and transitions that can be
-  performed on that data.
-* _Composable_ means that simple types can be grouped to describe complex data structures.
-  Microstates knows how to transition complex data structures in an immutable way.
-* _Immutable_ means that state in the microstate cannot be modified. To change the state, you must invoke a transition.
-* _Transition_ is an operation that has access to the current state, might perform computations/transformations on it and returns a new state.
+* 🍇 Composable type system
+* 🍱 Reusable state atoms
+* 💎 Pure immutable state transitions without writing reducers
+* ⚡️ Lazy and synchronous out of the box
+* 🦋 Most elegant way to express state machines
+* 🎯 Transpilation free type system
+* 🔭 Optional integration with Observables
+* ⚛ Use in Node.js and browser
 
-To add microstates to your project,
+But, most imporantly, Microstates makes working with state fun.
 
-```bash
-npm install --save microstates
+**When was the last time you had fun working with state?**
 
-# yarn add microstates
-```
+For many, the answer is probably never, because state management in JavaScript is an endless game of compromises. You can choose to go fully immutable and write endless reducers. You can go mutable and everything becomes an observable. Or you can `setState` and loose the benefit of serialization and time travel debugging.
 
-Let's see some code,
+Unlike the view layer, where most frameworks agree on some variation of React components, state management is a lot less certain. It's less certain because none of the available tools strike the balance that React components introduced to the view layer.
```

Comment:
`variation of React components` - would it be more accurate to say something like `variation of React's concept of components`?  I'm guessing most frameworks other than React aren't using actual React components.

Reply:
`...is a lot less certain. It's less certain...` - echo.  can change the opening of second sentence to something like `This is because` or join the sentences into one.  The former is probably preferable: joining would make this a bit long.